src/main/java/com/autotune/analyzer/services/Summarize.java

```diff
@@ -0,0 +1,915 @@
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.autotune.analyzer.utils.AnalyzerConstants.ServiceConstants.CHARACTER_ENCODING;
+import static com.autotune.analyzer.utils.AnalyzerConstants.ServiceConstants.JSON_CONTENT_TYPE;
+
+public class Summarize extends HttpServlet {
+    private static final Logger LOGGER = LoggerFactory.getLogger(Summarize.class);
+    HashMap<String, SummarizeAPIObject> clusterSummaryCacheMap = new HashMap<>();
+    HashMap<String, SummarizeAPIObject> namespaceSummaryCacheMap = new HashMap<>();
+
+    @Override
+    public void init(ServletConfig config) throws ServletException {
+        super.init(config);
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Timer.Sample timerListRec = Timer.start(MetricsConfig.meterRegistry());
+        response.setStatus(HttpServletResponse.SC_OK);
+        response.setContentType(JSON_CONTENT_TYPE);
+        response.setCharacterEncoding(CHARACTER_ENCODING);
+        List<SummarizeAPIObject> summarizeAPIObjectList = new ArrayList<>();
+        String summarizeType = request.getParameter(KruizeConstants.JSONKeys.SUMMARIZE_TYPE);
+        String clusterName = request.getParameter(KruizeConstants.JSONKeys.CLUSTER_NAME);
+        String namespaceName = request.getParameter(KruizeConstants.JSONKeys.NAMESPACE_NAME);
+        String fetchFromDB = request.getParameter(KruizeConstants.JSONKeys.FETCH_FROM_DB);
+
+        // validate Query params
+        Set<String> invalidParams = new HashSet<>();
+        for (String param : request.getParameterMap().keySet()) {
+            if (!KruizeSupportedTypes.SUMMARIZE_PARAMS_SUPPORTED.contains(param)) {
+                invalidParams.add(param);
+            }
+        }
+        if (invalidParams.isEmpty()) {
+            // Set default value if absent
+            if (summarizeType == null || summarizeType.isEmpty())
+                summarizeType = KruizeConstants.JSONKeys.CLUSTER;
+            // by default, db_flag will be false so the data will be fetched from cache only
+            if (fetchFromDB == null || fetchFromDB.isEmpty())
+                fetchFromDB = AnalyzerConstants.BooleanString.FALSE;
+            if (isValidValue(summarizeType, fetchFromDB)) {
+                // load recommendations based on params
+                try {
+                    if (fetchFromDB.equals(AnalyzerConstants.BooleanString.TRUE)) {
+                        clusterSummaryCacheMap = new HashMap<>();
+                        namespaceSummaryCacheMap = new HashMap<>();
+                    }
+                    // get the clusters and namespaces
+                    HashMap<String, List<String>> namespacesWithClustersMap = new ExperimentDBService().loadNamespacesByClusterNames();
```

Comment:
rename variable name as
ClusterNamespaceAssociationMap

and ExperimentDBService().loadAllClusterNamespaceAssociationMap

Reply:
Done